src/dmd/doc.d

```diff
@@ -1920,6 +1920,74 @@ unittest {
     assert(skipChars(&buf, 12, "dne") == 15);
 }
 
+/****************************************************
+ * Escape the given string
+ * Params:
+ *  s = the string to escape
+ *  c = the character to look for
+ *  r = the string to replace `c` with
+ * Returns: `s` with `c` replaced with `r`
+ */
+private string escapeString(string s, char c, string r) pure
+{
+    for (size_t i = 0; i < s.length; ++i)
```

Comment:
It's very likely faster to count the occurrences of `c` first, then allocate the result array once and then fill it.

Reply:
Made that change.